stl/inc/algorithm

```diff
@@ -1565,6 +1582,158 @@ _NODISCARD _CONSTEXPR20 _FwdItHaystack search(
     return _Search(_First, _Last).first;
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // FUNCTION TEMPLATE _Equal_rev_pred
+    // clang-format off
+    template <class _It1, class _It2, class _Se2, class _Pr, class _Pj1, class _Pj2>
+    concept _Equal_rev_pred_can_memcmp = is_same_v<_Pj1, identity> && is_same_v<_Pj2, identity>
+        && is_same_v<_Se2, _It2> && _Equal_memcmp_is_safe<_It1, _It2, _Pr>;
+
+    template <input_iterator _It1, input_iterator _It2, sentinel_for<_It2> _Se2, class _Pr, class _Pj1, class _Pj2>
+        requires indirectly_comparable<_It1, _It2, _Pr, _Pj1, _Pj2>
+    _NODISCARD constexpr pair<bool, _It1> _Equal_rev_pred(
+        _It1 _First1, _It2 _First2, const _Se2 _Last2, _Pr _Pred, _Pj1 _Proj1, _Pj2 _Proj2) {
+        // Returns {true, _First1 + (_Last2 - _First2)} if [_First1, ...) equals [_First2, _Last2), and {false, {}}
+        // otherwise.
+        constexpr bool _Optimize = _Equal_rev_pred_can_memcmp<_It1, _It2, _Se2, _Pr, _Pj1, _Pj2>;
+        if constexpr (_Optimize) {
+            if (!_STD is_constant_evaluated()) {

```

Comment:
We can't merge these together since the outer is `if constexpr`. `_Optimize` must be a constant expression, and evaluating `is_constant_evaluated()` in a manifestly constant expression is always a bug since it's necessarily `true`.